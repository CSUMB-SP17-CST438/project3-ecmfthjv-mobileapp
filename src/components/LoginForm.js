import React, { Component } from 'react';
import { View } from 'react-native';
import { Container, Content, Form, Item, Input, Label, Button, Text, Card } from 'native-base';
import { FormInput } from './FormInput';
import firebase from 'firebase';

class LoginForm extends Component {
  state = {
    email: '',
    password: ''
  };

  onButtonPress() {
    const { email, password } = this.state;
      firebase.auth().signInWithEmailAndPassword(email, password);
  }

  render() {
        return (
                <Content>
                  <Form>
                  <Card>
                      <FormInput
                          labelProp='email'
                          value={this.state.email}
                          onChangeText={ email => this.setState({ email }) }
                      />
                  </Card>
                  <Card>
                      <FormInput
                          labelProp='password'
                          value={this.state.password}
                          onChangeText={ password => this.setState({ password }) }
                      />
                  </Card>
                        <Card>
                            <Button full success onPress={this.onButtonPress.bind(this)}>
                                <Text>Log In</Text>
                            </Button>
                        </Card>
                    </Form>
                </Content>
        );
    }
}
export default LoginForm;
