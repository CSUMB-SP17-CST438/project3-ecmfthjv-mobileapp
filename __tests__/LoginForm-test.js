import 'react-native';
import React from 'react';
import LoginForm from '../src/components/LoginForm';

import renderer from 'react-test-renderer';

test('LoginForm view renders correctly', () => {
  const tree = renderer.create(
    <LoginForm />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
