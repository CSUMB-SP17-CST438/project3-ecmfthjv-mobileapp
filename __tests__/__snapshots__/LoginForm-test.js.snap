// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm view renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  onScroll={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      style={Object {}}
      virtual={undefined}
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#ccc",
            "borderRadius": 2,
            "borderWidth": 0.5,
            "elevation": 3,
            "flex": 1,
            "flexWrap": "wrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": 1,
              "flexDirection": "row",
              "marginLeft": 2,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "left": 15,
                "opacity": 1,
                "paddingBottom": undefined,
                "paddingTop": undefined,
                "position": "absolute",
                "right": 0,
                "top": 18,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#575757",
                  "fontSize": 17,
                  "paddingRight": 5,
                  "top": 8,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 17,
                  }
                }
                virtual={undefined}
              >
                email
              </Text>
            </Text>
          </View>
          <TextInput
            autoCorrect={false}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="#575757"
            secureTextEntry={undefined}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 60,
                "lineHeight": 24,
                "paddingLeft": 5,
                "paddingRight": 5,
                "top": 8,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            value=""
            virtual={undefined}
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#ccc",
            "borderRadius": 2,
            "borderWidth": 0.5,
            "elevation": 3,
            "flex": 1,
            "flexWrap": "wrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": 1,
              "flexDirection": "row",
              "marginLeft": 2,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "left": 15,
                "opacity": 1,
                "paddingBottom": undefined,
                "paddingTop": undefined,
                "position": "absolute",
                "right": 0,
                "top": 18,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#575757",
                  "fontSize": 17,
                  "paddingRight": 5,
                  "top": 8,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 17,
                  }
                }
                virtual={undefined}
              >
                password
              </Text>
            </Text>
          </View>
          <TextInput
            autoCorrect={false}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="#575757"
            secureTextEntry={true}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 60,
                "lineHeight": 24,
                "paddingLeft": 5,
                "paddingRight": 5,
                "top": 8,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            value=""
            virtual={undefined}
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#ccc",
            "borderRadius": 2,
            "borderWidth": 0.5,
            "elevation": 3,
            "flex": 1,
            "flexWrap": "wrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#5cb85c",
              "borderColor": "#007aff",
              "borderRadius": 0,
              "borderWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 6,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "lineHeight": 19,
                "marginLeft": 0,
                "marginRight": 0,
              }
            }
            virtual={undefined}
          >
            Log in
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#62B1F6",
              "borderColor": "#007aff",
              "borderRadius": 0,
              "borderWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 6,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "lineHeight": 19,
                "marginLeft": 0,
                "marginRight": 0,
              }
            }
            virtual={undefined}
          >
            Log in with Google
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
